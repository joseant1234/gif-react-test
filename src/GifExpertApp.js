import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['Naruto', 'Caballeros del zodiaco', 'Dragon ball']
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     // setCategories([...categories, 'Anime']);
    //     setCategories(cats => [...cats, 'Anime']);
    // }

    return (
        <>
            <h2>GiExpertApp</h2>

            <AddCategory setCategories={setCategories}></AddCategory>
            <hr/>

            <ol>
                {
                    categories.map(category =>
                        <GifGrid
                            category={category}
                            key={category}
                        />
                    )
                }
            </ol>
        </>
    )
}
