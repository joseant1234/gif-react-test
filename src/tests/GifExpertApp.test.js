import React from 'react';
import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';

describe('Test in <GifExpertApp />', () => {
    test('should show correctly', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show a category list', () => {
        const categories = ['Naruto'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
})
