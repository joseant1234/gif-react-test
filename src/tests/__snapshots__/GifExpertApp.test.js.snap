// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifExpertApp /> should show a category list 1`] = `
<Fragment>
  <h2>
    GiExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`Test in <GifExpertApp /> should show correctly 1`] = `
<Fragment>
  <h2>
    GiExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
